conda_libmamba_solver/solver.py

```diff
@@ -757,13 +757,13 @@ def _notify_conda_outdated(
             return
 
         channel_name = current_conda_prefix_rec.channel.canonical_name
-        if channel_name == UNKNOWN_CHANNEL:
+        if channel_name in (UNKNOWN_CHANNEL, "@", "<develop>", "pypi"):
             channel_name = "defaults"
 
         # only check the loaded index if it contains the channel conda should come from
         # otherwise ignore
         index_channels = {getattr(chn, "canonical_name", chn) for chn in index._channels}
-        if channel_name in index_channels:
+        if channel_name not in index_channels:
```

Comment:
Apparently we had this logic reversed 😨 

This now will indeed check where there's a conda update if the solution includes conda, and we happened to have the channel conda came from in the loaded channels already (to prevent additional fetching).

Reply:
Yikes, that seems bad, will take a closer look at what that means..